03_letters/nicholeL/main_test.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"bytes"
+	"reflect"
+	"strconv"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+	main()
+	excepted := strconv.Quote("a:2\nb:1\n")
+	actual := strconv.Quote(buf.String())
+	if excepted != actual {
```

Comment:
Another great choice is using the [assert.Equal package](https://godoc.org/github.com/stretchr/testify/assert) package, and will save you lines of code in future labs. Plus it prints out nice messages if tests fails.

Reply:
thanks shen, will update